frontend/src/metabase/dashboard/containers/DashboardApp.jsx

```diff
@@ -18,7 +18,12 @@ import { useWebNotification } from "metabase/hooks/use-web-notification";
 import { useOnUnmount } from "metabase/hooks/use-on-unmount";
 
 import { fetchDatabaseMetadata } from "metabase/redux/metadata";
-import { getIsNavbarOpen, setErrorPage } from "metabase/redux/app";
+import {
+  getIsNavbarOpen,
+  setErrorPage,
+  setCollectionId,
```

Comment:
I believe it was incorrect to store `collectionId` in the redux state in the first place. We should have defined it as a selector. Same for `showBreadcrumbs`. Can this be changed in this PR?

Reply:
We can certainly make a change here, and I expect it to change quite a bit when we handle the "forked question" use case as well. The Dashboard looks like it resets part of it's state when you navigate away from the page. I can also do something similar in the query builder